### DISCUSSION TITLE
Exploratory Guidelines and Outcomes Draft

### DISCUSSION THREAD

**Discussion by rachaelbradley** [w3c/wcag3#60] [General]
At the [26 March 2024 meeting](https://www.w3.org/2024/03/26-ag-minutes#t03), we introduced a [draft list of outcomes and guidelines](https://docs.google.com/document/d/1mxIImo7FgjyDzEu6J5jWvHIDFC_07TOR46Hn4hIAQvU/edit#heading=h.l86m24tnln43) based on the work we've been doing.   Please review this and add comments directly to the document or here in GitHub. Also feel free to suggest alternatives.

**Comment by rachaelbradley** [comment]
I've taken a pass at revising and slightly reorganizing the list based on comments. I've also tried to make the language consistent. [Draft outcomes and guidelines alternative](https://docs.google.com/document/d/1XA5X9f9G5vmKJ7BiC7GCf8nWaCjnKGcSYrFd9ImWxc8/edit?pli=1)

**Reply by bruce-usab** [reply]
Thanks for this reorganization!  I concur that it is a significant improvement from the previous draft.  This version makes it easier grasp the scope of what might be addressed.

**Comment by alastc** [comment]
(Chair hat off) Quite a few of the outcomes include "programmatic", which I think leads to an author-oriented method.

Can we remove that, so for this example:
> Error association: Error notifications are **programmatically** associated with the error source.

It is:
> Error association: Error notifications are associated with the error source.

It is more ambiguous, in that it doesn't hint at the solution or how you associate the message with the source. But, that is what allows it to apply whatever the mechanism is.

To use parlance I've heard from Rachael, the Outcome is a "verifiable statement", and we'll need something normative underneath this level as well for testable statements.

**Comment by rachaelbradley** [comment]
### Summary for 9 April 2024 Meeting

Below are some initial questions from the work so far that we ask everyone consider while looking at the draft:
- Does the grouping work for a placeholder exploratory draft?
- Are there any duplicates or ones that should be combined?
- Are there any missing?
- Should we note items that would have essential exceptions?

An important discussion has come up from this review. We will start the conversation at today's AG meeting with a discussion of how specific should outcomes be.
-- What exactly do we mean by “testable?” See [Writing Testable Outcomes](https://docs.google.com/document/d/1sugAtqie_x1XqHDZo1Im7ftDNllWeRV_ty4PULeoTV0/edit#heading=h.ekutz99zocwa) as a starting point
--- Extremely testable examples but also technology and implementation specific: [ACT](https://www.w3.org/WAI/standards-guidelines/act/rules/)
-- How do we balance “testable” and “future proofing”?
-- What is the scope that "testing" occurs in?

**Reply by rachaelbradley** [reply]
### Notes from Meeting
- We discussed [how specific to make outcomes](https://docs.google.com/document/d/1G1z1oU-XqWpIaoQAFPLvZuyTuF1DKRGF5_6H5oFI328/edit#heading=h.7jzuts8bgfta) both long term and for this exploratory draft. 
- We didn't reach a decision but the discussion was trending towards making decisions about specificity on an outcome by outcome basis.

**Comment by rachaelbradley** [comment]
### 10 April 2024
For this week, please continue reviewing and commenting on [Draft outcomes and guidelines alternative](https://docs.google.com/document/d/1XA5X9f9G5vmKJ7BiC7GCf8nWaCjnKGcSYrFd9ImWxc8/edit?pli=1) with particular focus on the following questions:
-- Does the grouping work for a placeholder exploratory draft?
-- Are there any duplicates or ones that should be combined?
-- Are there any missing outcomes?
-- Should we note items that would have essential exceptions?

**Comment by rachaelbradley** [comment]
### Summary for 16 April Meeting

**For Discussion in Meeting**
- Finalize conversation from last week. Suggested solution - For right now, keep visually and programmatically in the outcome. Evaluate each on a case by case basis as we work on them. Start a writing guide to work out and document this type of decision as we work. 
- We need to decide if its ok to leave out essential exceptions until the next stage of subgroup work. 
- Decide between process, task, or clarify when to use each
- Discussion is needed on "Visible error: Errors notifications are visually colocated with the source of the error within the viewport, or provide a link to the source of the error which moves the viewport." and "Error identification: Errors are visually identifiable without relying on only text, only color, or only symbols."  Questions on these:
-- Is an error identified with text only sufficient?
-- There is possibly some unintentional overlaps. What wording changes are needed to ensure these are independent and parallel?


**For Review** (Each bullet below has a comment to up or down vote.)
- Suggestion to change audio shifting from "Audio shifting: Audio shifting designed to create a perception of motion is avoided; or can be paused or prevented" to "Important information is not presented in spatial audio alone." and "Important information perception does not require depth perception."
- Can we remove "Obstructions: Users are not obstructed from completing tasks or accessing information (essential exception ex: Security)" as a duplicate of "Avoid obstructions: Tasks can be completed without navigating obstructions or redirections (the opposite of being manipulated is being in control)"  Can we also reword Avoid Obstructions to "Avoid manipulation:  Tasks can be completed without navigating misinformation or redirections." This better reflects the scratchpad work and clarifies obstructions.
- Can we remove "optional" from "Optional information: Tasks can be completed without entering unnecessary or optional information"?
- Can we remove "Known interaction types: The tasks available in a view can be understood without needing to learn new interactions." ? The intent based on the scratchpad seems to duplicate "Behavior of controls: Controls and inputs with the same functionality behave consistently."
- Can we remove "Color in media alternatives: Media alternatives include information conveyed by color alone." as a duplicate of "Use of color: Color alone does not convey meaning." and reword it to "Use of color: Information is not conveyed with color alone."? 
- Can we expand "Sound in media alternatives: Media alternatives include information conveyed by sound alone." to "Use of Sound: Information is not conveyed with sound alone."
- Can we remove "Multistep process: Provides context that orients the user in a site or multi-step process." as a duplicate of "Current location: The current location within the view, multi-step process, and aggregate is visually and programmatically indicated."
- Can we change "Haptic stimulation: Haptic feedback can be controlled" to "Haptic Stimulation: Haptic feedback can be **reduced or turned off**"?
- Can we change "3rd party content presentation: The presentation of advertising and other third-party content that obscures or interferes with the primary content can be controlled." to  "Advertising and other third-party content that obscures the primary content can be moved or removed before interacting with it."?
- Can we change "Control interruptions: The timing and positioning of notifications and other interruptions can be controlled, suppressed or saved, except interruptions involving an emergency." to "Control interruptions: The timing and positioning of notifications and other interruptions can be **changed**, suppressed or saved, except interruptions involving an emergency."

**Reply by rachaelbradley** [reply]
### Notes from Meeting
There was general consensus for:
- Keeping visually and programmatically in the outcome. Evaluate each on a case by case basis as we work on them. Start a writing guide to work out and document this type of decision as we work.
- Adding general framing around the need to add essential exceptions 
- Keeping both task and process, possibly also using action. Getting definitions for each

**Action Needed:**
- Please review each of the comments below. Each comment suggests a change. 
- Thumbs up if you agree with the change. 
- Thumbs down if you do not agree with the change. If you mark thumbs down add a comment replying to the individual comment explaining why. Please make sure you reply to the comment and not the general thread.
- Eyes only if you reviewed but were not sure whether the change was ok or not.

**Comment by rachaelbradley** [comment]
Change audio shifting from "Audio shifting: Audio shifting designed to create a perception of motion is avoided; or can be paused or prevented" to "Important information is not presented in spatial audio alone." and "Important information perception does not require depth perception."

**Reply by mraccess77** [reply]
This seems to change the intent of the audio shifting - seems like 2 needs - not just spatial audio & the need to not cause the sense of movement with audio.  

Also the need for no depth perception should be for visual use without depth perception as we wouldn't want to say you can do something without depth perception through audio as an example.  That would be a separate from this. The goal is for visual access without depth perception.

**Reply by rachaelbradley** [reply]
Conversation from Word document and Github:
- Several members suggested removing this from the list as it is controlled by the user through the system and not by the content creator.  Questions to consider: Are we certain that VR and other future technologies will not allow the content creator to control the audio. Are we intentionally limiting this list at this stage to information controlled by the content creator?
- Do we need to add "Important information is not presented in spatial audio alone." and "Important information perception does not require depth perception."

**Reply by mraccess77** [reply]
In my opinion requiring the user to switch to or use mono audio at the system level to avoid any possible audio motion seems like an unfair request. How would the user the know to do this ahead of time? How could the user get the benefits of stereo audio without getting motion?

**Comment by rachaelbradley** [comment]
Remove "Obstructions: Users are not obstructed from completing tasks or accessing information (essential exception ex: Security)" as a duplicate of "Avoid obstructions: Tasks can be completed without navigating obstructions or redirections (the opposite of being manipulated is being in control)"  AND Reword Avoid Obstructions to "Avoid manipulation: Tasks can be completed without navigating misinformation or redirections." 

This better reflects the scratchpad work and clarifies obstructions.

**Comment by rachaelbradley** [comment]
Remove "optional" from "Optional information: Tasks can be completed without entering unnecessary or optional information". 

It would then read "Unnecessary information: Tasks can be completed without entering unnecessary information

**Comment by rachaelbradley** [comment]
Remove "Known interaction types: The tasks available in a view can be understood without needing to learn new interactions."

The intent based on the scratchpad seems to duplicate "Behavior of controls: Controls and inputs with the same functionality behave consistently." and "Consistent labels: Controls and inputs with the same functionality have consistent labels."

**Comment by rachaelbradley** [comment]
Remove "Color in media alternatives: Media alternatives include information conveyed by color alone." as a duplicate of "Use of color: Color alone does not convey meaning." AND reword it to "Use of color: Information is not conveyed with color alone."

**Comment by rachaelbradley** [comment]
Expand "Sound in media alternatives: Media alternatives include information conveyed by sound alone." to "Use of Sound: Information is not conveyed with sound alone."

**Comment by rachaelbradley** [comment]
Remove "Multistep process: Provides context that orients the user in a site or multi-step process." as a duplicate of "Current location: The current location within the view, multi-step process, and aggregate is visually and programmatically indicated."

**Reply by julierawe** [reply]
COGA's Clear Language team is concerned about this suggestion. As currently worded, the "Current location" outcome is too narrow to be considered a duplicate of the "Multi-step process" outcome. 

Multi-step processes need to provide context that goes beyond indicating current location. For example, it's essential for multi-step processes to give users a sense of how many steps are in the process, how many have been completed, what the current step is and, if applicable, what the next step is and whether there are time limits for completing each step. Multi-step processes need to make clear not just where you are, but what your overall goal is and how you'll get there.

**Reply by deljennie-a11y** [reply]
I think what @julierawe has identified - the key different between the text in quotes in @rachaelbradley's post is the word presence and absence of "context." This is important. Is there a need for 2 outcomes instead of 1? 
Example:
- One outcome could reference current location within a view, multi-step process... Essentially, the "you are here."
- Another outcome could reference current location in relationship to a process - the context. This could include: knowing which step you are on in a multi-step process and total number of steps, if there are time-limits, goal, how to progress.

**Comment by rachaelbradley** [comment]
Change "Haptic stimulation: Haptic feedback can be controlled" to "Haptic Stimulation: Haptic feedback can be reduced or turned off"

**Comment by rachaelbradley** [comment]
Change "3rd party content presentation: The presentation of advertising and other third-party content that obscures or interferes with the primary content can be controlled." to "Advertising and other third-party content that obscures the primary content can be moved or removed before interacting with it."

**Comment by rachaelbradley** [comment]
Change "Control interruptions: The timing and positioning of notifications and other interruptions can be **controlled**, suppressed or saved, except interruptions involving an emergency." to "Control interruptions: The timing and positioning of notifications and other interruptions can be **changed**, suppressed or saved, except interruptions involving an emergency."

**Comment by rachaelbradley** [comment]
Change "Decorative indicator: Decorative and informative images are clearly indicated " to Image role: The role and importance of images are programmatically indicated. 

This change was based on discussion in group work on text alternatives.

**Comment by rachaelbradley** [comment]
Break apart text minimum into:
**Text minimum**: The rendered text meets a minimum font size and weight. 
**Text style:** The rendered text and does not use a decorative or cursive font face.

**Comment by rachaelbradley** [comment]
### Summary for April 23rd Meeting

#62 presents these guidelines within the document. 

**Changes with approval here**
- Remove "Obstructions: Users are not obstructed from completing tasks or accessing information (essential exception ex: Security)" as a duplicate of "Avoid obstructions: Tasks can be completed without navigating obstructions or redirections (the opposite of being manipulated is being in control)" AND Reword Avoid Obstructions to "Avoid manipulation: Tasks can be completed without navigating misinformation or redirections."  
- Removed "optional" from "Optional information: Tasks can be completed without entering unnecessary or optional information".  It would then read "Unnecessary information: Tasks can be completed without entering unnecessary information. 
- Removed "Known interaction types: The tasks available in a view can be understood without needing to learn new interactions."  The intent based on the scratchpad seems to duplicate "Behavior of controls: Controls and inputs with the same functionality behave consistently." and "Consistent labels: Controls and inputs with the same functionality have consistent labels." 
- Expanded "Sound in media alternatives: Media alternatives include information conveyed by sound alone." to "Use of Sound: Information is not conveyed with sound alone." 
- Changed "Haptic stimulation: Haptic feedback can be controlled" to "Haptic Stimulation: Haptic feedback can be reduced or turned off"  
- Changed "3rd party content presentation: The presentation of advertising and other third-party content that obscures or interferes with the primary content can be controlled." to "Advertising and other third-party content that obscures the primary content can be moved or removed before interacting with it." 
- Changed "Control interruptions: The timing and positioning of notifications and other interruptions can be controlled, suppressed or saved, except interruptions involving an emergency." to "Control interruptions: The timing and positioning of notifications and other interruptions can be changed, suppressed or saved, except interruptions involving an emergency." 
- Change "Decorative indicator: Decorative and informative images are clearly indicated " to Image role: The role and importance of images are programmatically indicated.
-Break apart text minimum into:
--Text minimum: The rendered text meets a minimum font size and weight.
--Text style: The rendered text and does not use a decorative or cursive font face. 

**Additional changes made (need review)**
- We’d lost the outcome for double negatives during cutting and pasting out of the scratchpads. Added back in. 
- Changed “Identify autogenerated textIdentify AI: Auto generated text alternatives descriptions are identified” to “Identify autogenerated text: Auto generated text alternatives descriptions are identified”
- Added “non-image” to “Non-text alternatives: Text alternatives are available for non-text, non-image content that conveys context or meaning”
- Removed “(/prioritization)”: from “Control importance: The importance of controls is defined.”

**Items that need discussion**
- [Audio Shifting](https://github.com/w3c/wcag3/discussions/60#discussioncomment-9131345)
- [Multistep process](https://github.com/w3c/wcag3/discussions/60#discussioncomment-9131447)

**Comment by detlevhfischer** [comment]
> Task completion documentation: Detailed documentation on task completion is available

I think this would need a qualification, some thing like "For tasks where labels and instructions cannot provide sufficient instructions for completion, detailed documentation on task completion is available." Otherwise, you would require extra docuemtnation for the simplest things, like adding your contact details in a form. I guess that#s not what is intended?

**Comment by detlevhfischer** [comment]
> Provide a feedback mechanism

This seems overly prescriptive if applied to all tasks.

**Comment by detlevhfischer** [comment]
> Audio alternative in preferred language

Is this really an author requirement or some hint at technical wizardry of UA / OS (automatically translate to Chinese and send through speech engine)?

**Comment by detlevhfischer** [comment]
> Interactive audio alternative:

Not sure what exactly is intended or required here. Be abel to press Pause and look up term in another browser tab?

**Comment by detlevhfischer** [comment]
> Color in media alternatives

This seems more useful to cover in a generic way, as today in 1.4.1

**Comment by detlevhfischer** [comment]
> Non-verbal cues: A mechanism is available

What is meant with "mechanism"? Isn't it just the skill of the captioner to squeeze in as much nonverbal info as possible in the given time? So just an editorial issue?

**Comment by detlevhfischer** [comment]
> Use of color

This seems misplaced in a category "Image and media alternatives". It applies just as much to text.

**Comment by detlevhfischer** [comment]
> Supplements to text:  Visual illustrations, pictures, and images are available to help explain complex ideas, events, and processes.

I can't see how this would become a normative requirement. So the idea is you would fail, say, a paper in a scientific journal if it does not include diagrams?

**Comment by detlevhfischer** [comment]
> Control updates: Changes to control or input name, roles, values or states are indicated

"are indicated" seems a bit vague. Maybe something like "are visually and programmatically available"?

**Comment by detlevhfischer** [comment]
> Control importance: The importance(/prioritization) of controls is defined.

"is defined" is a bit unclear anfthe whole requirement seems too general. Maybe: "Where processes prioritise controls over others, the priority is visually and programmatically available." Would setting focus on "Cancel" rather than "Delete" in a dialog confirming deletion an example of this? Or what use case is addressed here?

**Comment by detlevhfischer** [comment]
> Pointer cancellation: Pointer cancellation is consistent.

I'd like something more specific here.

**Reply by alastc** [reply]
Hi Detlev, at the outcome level I'm not sure how much more specific we can get, but there would be (technology specific) methods under the outcome which are more specific.

From previous conversations I'm assuming that these are questions / comments that you're happy to keep discussing without blocking the update to the working draft? 

If you think it's urgent, please reply to the (upcoming) CFC, otherwise we can discuss after the publication.

**Reply by detlevhfischer** [reply]
No that's fine, not intent on blocking anything...

**Comment by detlevhfischer** [comment]
> Section length, Section headers, White spacing

Cannot apply to all sorts of content, seems far too prescriptive.

**Comment by detlevhfischer** [comment]
> Focus retention: A user can focus on a content “area” then resume their view of all content using a limited number of steps

I do not understand the use case. Are we talking about popups and returning the focus to the trigger when leaving / closing the popup? Then it should be more specific. Otherwse it is fairly woolly what is intended here and how it would be measured.

**Comment by detlevhfischer** [comment]
> Section labels

Seems a duplicate of Section headers. Remove?

**Comment by detlevhfischer** [comment]
>Text and Wording

Many of the editorially-oriented outcomes in section "Text and wording" appear to me too restrictive to be applied to content generally as a normative requirement.

**Comment by rachaelbradley** [comment]
I've taken a first pass at addressing these comments in https://github.com/w3c/wcag3/pull/63

I am closing this discussion and moving conversation to the pull requests.
